internal/sync/syncer_related_test.go

```diff
@@ -0,0 +1,367 @@
+
+	testcases := []testcase{
+		{
+			name:           "optional related resource does not exist",
+			remoteAPIGroup: "remote.example.corp",
+			localCRD:       loadCRD("things"),
+			pubRes: newPublishedResources([]syncagentv1alpha1.RelatedResourceSpec{
+				{
+					Identifier: "optional-secret",
+					Origin:     "service",
+					Kind:       "Secret",
+					Reference: syncagentv1alpha1.RelatedResourceReference{
+						Name: syncagentv1alpha1.ResourceLocator{
+							Path: "metadata.name",
+							Regex: &syncagentv1alpha1.RegexResourceLocator{
+								Replacement: "optional-credentials",
+							},
+						},
+					},
+				},
+			}),
+			performRequeues: true,
+			remoteObject: newUnstructured(&dummyv1alpha1.NamespacedThing{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-test-thing",
+					Namespace: stateNamespace,
+				},
+				Spec: dummyv1alpha1.ThingSpec{
+					Username: "Colonel Mustard",
+				},
+			}, withGroupKind("remote.example.corp", "RemoteThing")),
+			localObject: newUnstructured(&dummyv1alpha1.NamespacedThing{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testcluster-my-test-thing",
+					Namespace: stateNamespace,
+					Labels: map[string]string{
+						agentNameLabel:            "textor-the-doctor",
+						remoteObjectClusterLabel:  "testcluster",
+						remoteObjectNameHashLabel: "c346c8ceb5d104cc783d09b95e8ea7032c190948",
+					},
+					Annotations: map[string]string{
+						remoteObjectNameAnnotation:      "my-test-thing",
+						remoteObjectNamespaceAnnotation: stateNamespace,
+					},
+				},
+				Spec: dummyv1alpha1.ThingSpec{
+					Username: "Colonel Mustard",
+				},
+			}),
+			existingState: "",
+
+			expectedRemoteObject: newUnstructured(&dummyv1alpha1.NamespacedThing{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-test-thing",
+					Namespace: stateNamespace,
+					Finalizers: []string{
+						deletionFinalizer,
+					},
+				},
+				Spec: dummyv1alpha1.ThingSpec{
+					Username: "Colonel Mustard",
+				},
+			}, withGroupKind("remote.example.corp", "RemoteThing")),
+			expectedLocalObject: newUnstructured(&dummyv1alpha1.NamespacedThing{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testcluster-my-test-thing",
+					Namespace: stateNamespace,
+					Labels: map[string]string{
+						agentNameLabel:            "textor-the-doctor",
+						remoteObjectClusterLabel:  "testcluster",
+						remoteObjectNameHashLabel: "c346c8ceb5d104cc783d09b95e8ea7032c190948",
+					},
+					Annotations: map[string]string{
+						remoteObjectNameAnnotation:      "my-test-thing",
+						remoteObjectNamespaceAnnotation: stateNamespace,
+					},
+				},
+				Spec: dummyv1alpha1.ThingSpec{
+					Username: "Colonel Mustard",
+				},
+			}),
+			expectedState: `{"apiVersion":"remote.example.corp/v1alpha1","kind":"RemoteThing","metadata":{"name":"my-test-thing","namespace":"kcp-system"},"spec":{"username":"Colonel Mustard"}}`,
+		},
+		{
+			name:           "mandatory related resource does not exist",
```

Comment:
Is the "not" wrong? The mandatory secret does exist, no?

Reply:
yes, this is left-over :(